### PR TITLE
Android: Build with vulkan source files

### PR DESCRIPTION
**Pull request by gaochun**
Build with vulkan backend source files and add corlor format mapping from gbm to vulkan

### CODE REVIEW

**Comment by gaochun** [comment]
@strassek Pls help review the patches.

File: Android.mk [hunk 1]
```diff
@@ -98,7 +98,27 @@ ifeq ($(strip $(BOARD_DISABLE_NATIVE_COLOR_MODES)),true)
 LOCAL_CPPFLAGS += -DDISABLE_NATIVE_COLOR_MODES
 endif
 
-ifeq ($(strip $(BOARD_USES_VULKAN)),)
+ifeq ($(strip $(BOARD_USES_VULKAN)),true)
+LOCAL_SHARED_LIBRARIES += \
+        libvulkan
```
Comment (strassek) [inline Android.mk]: I think this file has been using tabs instead of spaces, might as well be consistent...

File: common/compositor/vk/vkshim.h [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #define VKSHIM_H_
 
 #include <drm/drm_fourcc.h>
+#include <gbm.h>
```
Comment (strassek) [inline common/compositor/vk/vkshim.h]: Is this needed outside of vkshim.cpp? Either way, It seems like this change belongs in the other patch...

Comment (kalyankondapally) [inline common/compositor/vk/vkshim.h]: Any gbm dependencies should be moved to os layer

Comment (kalyankondapally) [inline common/compositor/vk/vkshim.h]: Any gbm dependencies should be moved to os layer

Comment (gaochun) [inline common/compositor/vk/vkshim.h]: Compile failed without including gbm.h, I'm moving it to a better place.

Comment (kalyankondapally) [inline common/compositor/vk/vkshim.h]: You can add it to platformdefines.h and include that here also I see we are including gbm conversions here we need to figure out a way to isolate them

Comment (strassek) [inline common/compositor/vk/vkshim.h]: @gaochun Just had a chat about this with Kalyan, it seems like the most appropriate thing to do for now would be to rename this Vulkan format conversion function to something more generic (maybe NativeToVkFormat), and allow each platform to define their own version. So there would be version in os/linux/platformdefines.h that would do this GBM->VK conversion and a version in os/android/platfomrdefines.h that would do the appropriate thing for Android.

**Comment by kalyankondapally** [comment]
@gaochun Any update ?

File: common/compositor/vk/vkshim.h (hunk 2 above)
Comment (gaochun) [inline common/compositor/vk/vkshim.h]: I've renamed GbmToVkFormat to NativeToVkFormat and moved to platfomrdefines.h. But I found DrmToVkFormat was not used at any place , I think GbmToVkFormat should be replaced with DrmToVkFormat at https://github.com/intel/IA-Hardware-Composer/blob/master/wsi/drm/drmbuffer.cpp#L151,  right?

Comment (kalyankondapally) [inline common/compositor/vk/vkshim.h]: yes, makes sense

Comment (strassek) [inline common/compositor/vk/vkshim.h]: > GbmToVkFormat should be replaced with DrmToVkFormat

Almost, we should replace that with a call to NativeToVkFormat. There shouldn't be any need for DrmToVkFormat or GbmToVkFormat after this conversion.

File: common/compositor/vk/vkshim.cpp [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 #include "vkshim.h"
 
+#include <gbm.h>
```
Comment (kalyankondapally) [inline common/compositor/vk/vkshim.cpp]: remove this, add platformdefines.h

File: os/android/platformdefines.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "platformdefines.h"
+#include <drm/drm_fourcc.h>
+
+#ifdef USE_VK
+VkFormat NativeToVkFormat(int drm_format) {
```
Comment (kalyankondapally) [inline os/android/platformdefines.cpp]: this can be generic under os/

**Comment by gaochun** [comment]
@kalyankondapally @strassek I deleted my forked repo by mistake, so could not update the commit here, I've update the patch with a new PR https://github.com/intel/IA-Hardware-Composer/pull/108 , please review again.

**Pull request closed without merging**